pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -715,11 +715,11 @@ public void connectionOpened(final ClientCnx cnx) {
         }
 
         boolean isDurable = subscriptionMode == SubscriptionMode.Durable;
-        MessageIdData startMessageIdData;
+        MessageIdData startMessageIdData = null;
         if (isDurable) {
             // For regular durable subscriptions, the message id from where to restart will be determined by the broker.
             startMessageIdData = null;
-        } else {
+        } else if (startMessageId != null) {
```

Comment:
looks like in line 742 below, there is also a reference to `startMessageId`. Will it also throw NPE？

Reply:
Reader will use the code here, startMessageId can be obtained from Broker. 
However, a judgment was added to prevent scene triggers in the future.